src/Akka.Streams.Kafka/Stages/Consumers/CommitObservationLogic.cs

```diff
@@ -96,8 +96,8 @@ private void UpdateBatchForPartition(GroupTopicPartition groupTopicPartition, IC
                     OffsetBatch = OffsetBatch.Updated(dOffset);
                     break;
                 case CommittableOffsetBatch dOffsetBatch
-                    when dOffsetBatch.OffsetsAndMetadata.ContainsKey(groupTopicPartition)
```

Comment:
Another key bug fix: we were filtering on `OffsetsAndMetadata` here which is wrong - needed to filter on the regular `Offsets` values.